### ISSUE TITLE
Android: Activity reopens after calling Director::getInstance()->end();

### ISSUE THREAD

**Issue by milos1290** [cocos2d/cocos2d-x#13878]
On some devices, like Samsung Galaxy Note 4 when calling Director::getInstance()->end(); to exit game,
Activity will be recreated and game will be started again due to calling 

```
 android.os.Process.killProcess(android.os.Process.myPid());
```

This happens because system thinks that process was killed forcefully and tries to restore it back.

One solution could be to call Activity.finish(), and let the system take care of resources, or even System.exit(0);
I'm not sure about second one.

**Comment by felixjingga** [comment]
Yes, i can confirm this bug exists.. sometimes it restores the application, sometimes it won't.

**Comment by milos1290** [comment]
Yeah, this is not 100% reproducible bug.
